File: monitor.php
<?php

$settingFile = './settings.txt';
if (file_exists($settingFile) === false) {
    echo $settingFile , " is not found.\n";
    exit(1);
}

$settings = explode(PHP_EOL, file_get_contents($settingFile));

if (count($settings) - 1 !== 2) {
    echo 'Settings line numbers should be 2.';
    exit(1);
}

if (substr($settings[0], 0, 4) !== 'url=') {
    echo 'Setting should begin with url= in the first line.';
    exit(1);
}

if (substr($settings[1], 0, 24) !== 'ifttt_maker_service_url=') {
    echo 'Setting should begin with ifttt_maker_service_url= in the second line.';
    exit(1);
}

$monitoredUrl = substr($settings[0], 4);
$makerServiceUrl = substr($settings[1], 24);

$curl = curl_init($monitoredUrl);
$response = curl_exec($curl);
$responseInfo = curl_getinfo($curl);

if ($responseInfo['http_code'] === 200) {
    curl_close($curl);
    echo $monitoredUrl . ' healthy is good!';
    exit(0);
}
curl_close($curl);

$curl = curl_init($makerServiceUrl);
$json = [
    'date' => date('Y-m-d H:i:s'),
    'status' => $monitoredUrl . ' website is down!',
    'status_code' => $responseInfo['http_code'],
];

curl_setopt($curl, CURLOPT_POSTFIELDS, json_encode($json));
curl_setopt($curl, CURLOPT_HTTPHEADER, ['Content-Type:application/json']);
curl_exec($curl);
curl_close($curl);

File: monitor.py
import os
import sys
import json
import datetime
import urllib.request


setting_path = './settings.txt'
if os.path.isfile(setting_path) is False:
    print(setting_path + ' is not found.')
    sys.exit(1)


with open(setting_path, 'r') as file_handler:
    settings = file_handler.readlines()
    if len(settings) != 2:
        print('settings lenght should be 2')
        sys.exit(1)
    monitored_url = settings[0][0:-1]
    maker_service_url = settings[1][0:-1]

    if monitored_url[0:4] != 'url=':
        print('url setting should begin with url=')
        sys.exit(1)
    if maker_service_url[0:24] != 'ifttt_maker_service_url=':
        print('IFTTT maker service url should begin with ifttt_maker_service_url=')
        sys.exit(1)

    response = urllib.request.urlopen(monitored_url[4:])
    if response.status == 200:
        print(monitored_url[4:] + ' website healthy is good!')
        sys.exit(0)

    request = urllib.request.Request(maker_service_url[24:])
    request.add_header('Content-Type', 'application/json; charset=utf-8')
    json_payload = {
        'date': datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
        'status': monitored_url[4:] + ' website is down!',
        'status_code': response.status,
    }
    jsondata = json.dumps(json_payload)
    json_data_bytes = jsondata.encode('utf-8') 
    request.add_header('Content-Length', len(json_data_bytes))
    response = urllib.request.urlopen(request, json_data_bytes)
    resp_text = response.readlines()

    if response.status == 200:
        print(resp_text[0].decode('utf-8'))
